packages/ast-helpers/stringify.js

```diff
@@ -212,18 +212,18 @@ const es2017GeneratorJSX = {
   },
   // {...props}
   JSXSpreadAttribute(node, state) {
-    state.write('...(');
+    state.write(' {...');
     if (node.argument.type === 'LogicalExpression') {
       this[node.argument.left.type](node.argument.left, state);
-      state.write(' ');
+      state.write('(');
```

Comment:
Sorry for not explaining clearly. The `state.write('(');` should be above `this[node.argument.left.type](node.argument.left, state);` and the `state.write(')');` should be below `this[node.argument.right.type](node.argument.right, state);`, so it will wrap the whole expression in this way. And we should better keep these 2 `state.write(' ');` here because we still need the space between the props and the logical operator. So the whole if statement should be like:
```
if (node.argument.type === 'LogicalExpression') {
  state.write('(');
  this[node.argument.left.type](node.argument.left, state);
  state.write(' ');
  state.write(node.argument.operator);
  state.write(' ');
  this[node.argument.right.type](node.argument.right, state);
  state.write(')');
}
```
Or you can even combine the center 3 `state.write` together to be like:
```
if (node.argument.type === 'LogicalExpression') {
  state.write('(');
  this[node.argument.left.type](node.argument.left, state);
  state.write(` ${node.argument.operator} `);    // or state.write(' ' + node.argument.operator + ' '); 
  this[node.argument.right.type](node.argument.right, state);
  state.write(')');
}
```
The situation in this if statement is not very commonly seen, many generators don't even consider it, but it's great to keep it here. Hope this makes sense 🙂